addons/promise/fnc_done.sqf

```diff
@@ -0,0 +1,24 @@
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_promise_done
+
+Description:
+    Checks if a given promise was processed yet.
+
+Parameters:
+    _promiseHandle - Handle returned by CBA_fnc_promise_create
+
+Returns:
+    Boolean, true if it is done yet, false if it is not.
+
+Example:
+    (begin example)
+        _promiseHandle call CBA_fnc_promise_done;
+    (end)
+
+Author:
+    X39
+---------------------------------------------------------------------------- */
+#include "script_component.hpp"
+params ["_index", "_stamp"];
+private _promise = GVAR(requests) select 0;
+if (_promise isEqualTo 0) { true } else { _promise select 2 != _stamp }
```

Comment:
Missing newline @eof

Also could be simplified to:

```sqf
_promise isEqualTo 0 || {_promise select 2 != _stamp}
```
probably.

Reply:
Syntax error, missing then.
also if true then true else..
ugh